data/src/main/java/com/linkedin/data/schema/UnionDataSchema.java

```diff
@@ -36,6 +36,11 @@
  */
 public final class UnionDataSchema extends ComplexDataSchema
 {
+  /**
+   *
+   */
+  private boolean allowEmptyUnionResponse = false;
```

Comment:
The schema layer doesn't have the context about it being used in request/response. So using "response" here is not a good idea.

I'd suggest calling this flag "isPartialSchema" to indicate this is a partial schema created from projection. 
Update the doc to say "this flag indicates the union schema is a partial representation with a subset of members" And also say this is for internal use only.

Reply:
Ok